src/anyio/_backends/_asyncio.py

```diff
@@ -540,7 +540,10 @@ async def __aexit__(self, exc_type: Optional[Type[BaseException]],
 
         try:
             if len(exceptions) > 1:
-                raise ExceptionGroup(exceptions)
+                if all(isinstance(e, CancelledError) for e in exceptions):
+                    raise CancelledError
```

Comment:
if any of these `CancelledError`s have a message then the message should also be propagated